Doppler.BillingUser/Infrastructure/BillingRepository.cs

```diff
@@ -378,5 +392,375 @@ LEFT JOIN
+                @ccExpYear = encryptedCreditCard.ExpirationYear,
+                @ccHolderName = encryptedCreditCard.HolderName,
+                @date = DateTime.UtcNow,
+                @source = SourceTypeBuyCreditsId,
+                @accountingTypeDescription = AccountingEntryTypeDescriptionCCPayment,
+                @idAccountType = UserAccountType,
+                @idInvoiceBillingType = InvoiceBillingTypeQBL,
+                @accountEntryType = AccountEntryTypePayment,
+                @authorizationNumber = authorizationNumber,
+                @paymentEntryType = PaymentEntryTypePayment
+            });
+
+            return invoiceId;
+        }
+
+        public async Task<int> CreateBillingCreditAsync(AgreementInformation agreementInformation, UserBillingInformation user, UserTypePlanInformation newUserTypePlan)
+        {
+            var currentPaymentMethod = await GetCurrentPaymentMethod(user.Email);
+            var buyCreditAgreement = new CreateAgreement
+            {
+                IdUser = user.IdUser,
+                IdCountry = user.IdCountry,
+                IdPaymentMethod = (int)user.PaymentMethod,
+                IdCCType = currentPaymentMethod.IdCCType,
+                CCExpMonth = short.Parse(currentPaymentMethod.CCExpMonth),
+                CCExpYear = short.Parse(currentPaymentMethod.CCExpYear),
+                CCHolderFullName = currentPaymentMethod.CCHolderFullName,
+                CCIdentificationType = currentPaymentMethod.CCType,
+                CCIdentificationNumber = currentPaymentMethod.CCNumber,
+                CCNumber = currentPaymentMethod.CCNumber,
+                CCVerification = currentPaymentMethod.CCVerification,
+                IdConsumerType = !string.IsNullOrEmpty(currentPaymentMethod.IdConsumerType) ? int.Parse(currentPaymentMethod.IdConsumerType) : null,
+                RazonSocial = currentPaymentMethod.RazonSocial,
+                ResponsableIVA = user.ResponsableIVA,
+                Cuit = user.IdCountry == CountryEnumArgentina ? currentPaymentMethod.IdentificationNumber : null,
+                CFDIUse = user.CFDIUse,
+                PaymentWay = user.PaymentWay,
+                PaymentType = user.PaymentType,
+                BankName = user.BankName,
+                BankAccount = user.BankAccount
+            };
+
+            var now = DateTime.UtcNow;
+            buyCreditAgreement.BillingCredit = new BillingCreditModel()
+            {
+                Date = now,
+                PaymentDate = now,
+                ApprovedDate = now,
+                Approved = true,
+                Payed = true,
+                IdUserTypePlan = agreementInformation.PlanId,
+                PlanFee = (double?)agreementInformation.Total,
+                CreditsQty = newUserTypePlan.EmailQty ?? null,
+                ExtraEmailFee = newUserTypePlan.ExtraEmailCost ?? null
+            };
+
+            var utcNow = DateTime.UtcNow;
```

Comment:
Ya definiste `now`, creo que lo mejor es seguir usando el mismo valor.

```suggestion
            var now = DateTime.UtcNow;
            buyCreditAgreement.BillingCredit = new BillingCreditModel()
            {
                Date = now,
                PaymentDate = now,
                ApprovedDate = now,
                Approved = true,
                Payed = true,
                IdUserTypePlan = agreementInformation.PlanId,
                PlanFee = (double?)agreementInformation.Total,
                CreditsQty = newUserTypePlan.EmailQty ?? null,
                ExtraEmailFee = newUserTypePlan.ExtraEmailCost ?? null
            };
```

Reply:
Tenes razón, se me paso..., quedo así porque antes eran dos métodos que terminaron en uno solo. 
Lo saco.

Reply:
Listo, se cambio.